### PR TITLE
backport: trivial 2024 10 23 pr1

### PR DESCRIPTION
**Pull request by PastaPastaPasta**
## Issue being fixed or feature implemented
Batch of trivial backports

## What was done?
See commits

## How Has This Been Tested?
built locally; large combined merge passed tests locally

## Breaking Changes
Should be none

## Checklist:
  _Go over all the following points, and put an `x` in all the boxes that apply._
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have added or updated relevant unit/integration/functional/e2e tests
- [ ] I have made corresponding changes to the documentation
- [x] I have assigned this pull request to a milestone _(for repository code-owners and collaborators only)_

### CODE REVIEW

**Review by kwvg — DISMISSED** [review]
utACK 62c39973e83a2488fa2b36a1f24df782cf095123

File: test/functional/test_framework/crypto/bip324_cipher.py [hunk 1]
```diff
@@ -25,6 +25,8 @@ def pad16(x):
 
 def aead_chacha20_poly1305_encrypt(key, nonce, aad, plaintext):
     """Encrypt a plaintext using ChaCha20Poly1305."""
+    if plaintext is None:
```
Comment (UdjinM6) [inline test/functional/test_framework/crypto/bip324_cipher.py:28]: 29390: also a part of #6330 and it probably belongs there and should be dropped here

**Review by UdjinM6 — CHANGES_REQUESTED** [review]

File: test/functional/test_framework/crypto/bip324_cipher.py (hunk 1 above)
Comment (PastaPastaPasta) [reply test/functional/test_framework/crypto/bip324_cipher.py]: done

**Comment by PastaPastaPasta** [comment]
Rebased on latest develop to fix merge ff warning

File: depends/packages/bdb.mk [hunk 2]
```diff
@@ -13,7 +13,6 @@ $(package)_cflags+=-Wno-error=implicit-function-declaration -Wno-error=format-se
 $(package)_cxxflags+=-std=c++17
 $(package)_cppflags_freebsd=-D_XOPEN_SOURCE=600 -D__BSD_VISIBLE=1
 $(package)_cppflags_netbsd=-D_XOPEN_SOURCE=600
```
[the remark this answers is not in the document]
Comment (PastaPastaPasta) [reply depends/packages/bdb.mk:15]: Our openbsd docs are missing backports, unable to merge this PR relatively trivially atm

**Review by knst — APPROVED** [review]
utACK b70e091374ee1a5db4f8299e693eabd7df0a225e

File: src/test/README.md [hunk 3]
```diff
@@ -15,7 +15,8 @@ that runs all of the unit tests. The main source file for the test library is fo
 Unit tests will be automatically compiled if dependencies were met in `./configure`
 and tests weren't explicitly disabled.
 
-After configuring, they can be run with `make check`.
+After configuring, they can be run with `make check`, which includes unit tests from
+subtrees, or `make && make -C src check-unit` for just the unit tests.
```
[the remark this answers is not in the document]
Comment (PastaPastaPasta) [reply src/test/README.md:19]: Ahh, interesting!

[the remark this answers is not in the document]
Comment (knst) [reply src/test/README.md:19]: maybe actually add `test_dash_qt_check` to README.md ? but with or without it's fine

**Pull request merged**